api/v1beta1/spifilecontentrequest_types.go

```diff
@@ -0,0 +1,70 @@
+//
+// Copyright (c) 2021 Red Hat, Inc.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+type SPIFileContentRequestSpec struct {
+	FilePath string `json:"filePath"`
+	RepoUrl  string `json:"repoUrl"`
+	Ref      string `json:"ref,omitempty"`
+}
+
+type SPIFileContentRequestStatus struct {
+	Phase             SPIFileContentRequestPhase `json:"phase"`
+	LinkedBindingName string                     `json:"linkedBindingName"`
+	ErrorMessage      string                     `json:"errorMessage,omitempty"`
+	OAuthUrl          string                     `json:"oAuthUrl,omitempty"`
+	UploadUrl         string                     `json:"uploadUrl,omitempty"`
```

Comment:
`UploadUrl` of what? Can I upload the new file contents for the stuff that I'm trying to download here? Or is it for manually uploading the token if the phase is awaiting token data? I'm pretty sure it is the latter, but then again, I already know a thing or two about SPI. Please either clarify using code comments or name this better.